### ISSUE TITLE
guild/party new message notifications in menu bar can be blank until sync completes or stay blank

### ISSUE THREAD

**Issue by Alys** [HabitRPG/habitica#8207]
edit by Alys 2017-03-29, copying from [my comments below](https://github.com/HabitRPG/habitica/issues/8207#issuecomment-287724362):

In #8587 , @joedragons reported that this happens when a quest start message is put into party chat.

We've seen other reports that this is from quest start messages and I've noticed that once myself. For some reason when they result in notifications being sent, the party name isn't included. That would be a good place to start fixing this, even if it turns out that there are other causes as well (we can always open a new issue later if there are other causes after this one is closed).

-----

As reported by Snowdeath (e1d841df-0cc4-4d28-989b-4ebf8d92cf4c), cTheDragons (bc10ec41-4213-4482-bbb8-558e3c3ea8d5), and shanaqui (bb089388-28ae-4e42-a8fa-f0c2bfb6f779) in the Report a Bug guild, when your party or a guild has a new message, the notification for it can sometimes appear blank at first (screenshot below), and then after the sync is complete, the name of the party is filled in. However sometimes the name never appears and the notification stays blank (I've confirmed that in this case, this is not a notification from a deleted guild, which can produce the same effect).

![screenshot](https://i.imgur.com/GjrnHQZ.png)

I'm not sure how much can be done to help with this, and any fixes should wait until after an upcoming website improvement project, but I wanted to record this for reference if others report it. I'm also not sure how it happens since the group name is recorded in the `newMessages` object. ¯\_(ツ)_/¯ 

Here's the posts describing the problem (this starts part way through a long discussion in which Snowdeath gave us a lot of information to help us work out what was going wrong):

Snowdeath:
Once the notification appears, I refreshed to check if it was actually something else, and yes. It was a party notification!
 
Alys:
So just to confirm, you see a blank notification, you reload the page, and then the notification changes to one for your party? Well that's good to know, but definitely shouldn't be happening! Sounds like you were right about the website not being synced enough. It mustn't have loaded your party name yet. Thank you for the information!

Snowdeath:
Mhmm, that's exactly what happened. I know for a fact that it can be any notification (Party, Guild, etc). My guess is that I might have left Habitica's page up too long, or something?
 
Snowdeath:
The notifications hasn't stopped, but I noticed that it's likely from slow syncing because every single time I see the blank notification then click on a new tab or reload, it never fails to show me which actual notification I have. Maybe it's from idling on a page?

Alys:
Is the notification that starts out blank always for your party, or can it be from guilds too?

shanaqui:
Oh, I've been seeing this one! Yes, for me it's always for my party, and it goes away once the sync is complete, usually -- it just pops up blank while updating. By the time the push notification has gone, my in-Habitica notification has usually resolved itself.
 
cTheDragons:
I have noticed it too. It is to do with activity with the party page. (particular if there is quite large amount of activity with members who are past the 30 mark in the member list. ) Currently sitting in a party of 60.

Snowdeath:
I just checked, and my party is a total of exactly 30! (Including myself.) Yeah, it's an active group.

**Comment by Alys** [comment]
MEmry, the worker bee (df8b14f2-e135-4789-970d-5056e99fcc3f) has reported this in a party of three. I've edited the top post to remove the statement that it's only for large parties.

**Comment by Alys** [comment]
In https://github.com/HabitRPG/habitica/issues/8587 , @joedragons reported that this happens when a quest start message is put into party chat.

I wonder if it's from system messages in general - perhaps the title of the party hasn't been loaded at the time that a system message is put into party chat.

**Comment by Alys** [comment]
We've seen other reports that this is from quest start messages and I've noticed that once myself. For some reason when they result in notifications being sent, the party name isn't included. That would be a good place to start fixing this, even if it turns out that there are other causes as well (we can always open a new issue later if there are other causes after this one is closed).

**Comment by Alys** [comment]
I'm putting this on hold because we're most of the way through a project to change the website's front-end code and this might no longer happen afterwards, or if it does, there'll be new code that will need to be modified to fix it.

**Comment by Alys** [comment]
I'm closing this now that the new website has been released. I don't believe this happens any more but it can be reopened if I'm wrong.

**Comment by joedragons** [comment]
FYI, I've seen "this" once since the revamp.  What I'd say is the new behavior occurs as the screenshot below.  I haven't seen it more than once (and forgot about this issue report) so hadn't reported it.
Interestingly, the time I saw it was during a holiday so there were definitely no changes to the code (per the commit list of github) so it was truely intermittent.  Nothing says that this hasn't been fixed since then either, that's the problem with intermittant issues, as I don't doubt you know=)  If I see a reproducable case, I'll likely report in guild rather than here.  But FYI.

![habiticarevampfeedback15](https://user-images.githubusercontent.com/9559205/32837715-877e7eaa-c9dc-11e7-8b7c-f7e61064462f.png)

**Comment by Alys** [comment]
@joedragons Thanks for that! I think for the moment we'll assume that it might have been fixed, since our code is churning pretty quickly these days. :)

However if anyone does see that from now on, please do report it in the Report a Bug guild! We'd probably open a new issue for it, despite what I said above about reopening this one. :)
